server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/PdbDataManager.java

```diff
@@ -46,82 +56,140 @@
-            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-
-            StringReader reader = new StringReader(rcsbpbdData);
-            Object uniprotObject = unmarshaller.unmarshal(reader);
+            String query = replaceHttpSymbols("{entry(entry_id:\"%s\"){struct{title}struct_keywords{pdbx_keywords}" +
+                    "polymer_entities{uniprots{rcsb_uniprot_entry_name}rcsb_polymer_entity{pdbx_description}" +
+                    "rcsb_polymer_entity_container_identifiers{auth_asym_ids}}}}", pdbIds);
 
-            Dataset fetchedDatasetData = null;
-            if (uniprotObject instanceof Dataset) {
-                fetchedDatasetData = (Dataset) uniprotObject;
-            }
-
-            return fetchedDatasetData;
-        } catch (JAXBException e) {
+            return parseToDataset(QueryUtils.execute(rcsbApi.getDataset(query)), pdbIds);
+        } catch (RSCBResponseException e) {
             throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
-                    .ERROR_UNEXPECTED_FORMAT, location), e);
+                    .ERROR_UNEXPECTED_FORMAT, LOCATION), e);
         }
     }
 
-
     /**
-     *
      * @param pdbIds PDB id
      * @return data from query
      * @throws ExternalDbUnavailableException
      */
     public Dasalignment fetchPdbMapEntry(final String pdbIds) throws ExternalDbUnavailableException {
+        try {
+            String query = replaceHttpSymbols("{entry(entry_id:\"%s\"){polymer_entities{polymer_entity_instances" +
+                    "{rcsb_polymer_instance_feature{feature_positions{beg_seq_id end_seq_id}}rcsb_id}}}}", pdbIds);
 
-        ParameterNameValue[] params = new ParameterNameValue[]{new ParameterNameValue("query", pdbIds)};
+            return parseToDasalignment(QueryUtils.execute(rcsbApi.getDasalignment(query)));
 
-        String location = RCSB_SERVER + RCSB_PDP_MAP_TOOL + "?";
+        } catch (RSCBResponseException e) {
+            throw new ExternalDbUnavailableException(MessageHelper.getMessage(MessagesConstants
+                    .ERROR_UNEXPECTED_FORMAT, LOCATION), e);
+        }
+    }
 
-        String rcsbpbdData = httpDataManager.fetchData(location, params);
+    private String replaceHttpSymbols(String query, String pdbIds) {
+        return String.format(query, pdbIds)
+                .replaceAll("\\{", "%7B")
+                .replaceAll("}", "%7D")
+                .replaceAll(" ", "%20");
+    }
 
-        try {
-            JAXBContext jaxbContext = JAXBContext
-                    .newInstance("com.epam.catgenome.manager.externaldb.bindings.ecsbpdbmap");
-            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+    private Dasalignment parseToDasalignment(DasalignmentDTO das) {
+        List<Alignment> alignments = new ArrayList<>();
+
+        List<PolymerEntity> polymerEntities = das.getData().getEntry().getEntities();
```

Comment:
Such `for-loop` can be easily rewrote with StreamAPI.
let's rewrite it with stream API, at least we can get rid of intermediate links, f.e. `pdbBlocks`, `segments` ..

Reply:
Fixed, as possible!

Reply:
let's make all intermediate links - `final`. Try to use final where it is possible

Reply:
fixed!